packages/cmr-client/src/ingestConcept.ts

```diff
@@ -50,17 +46,28 @@ async function ingestConcept(
       }
     );
 
-    const ingestResponseBody = <IngestResponseErrors>(await parseXMLString(response.body));
+    return <CMRResponseBody>(await parseXMLString(response.body));
+  } catch (error) {
+    log.error(error, logDetails);
+
+    const statusCode = get(error, 'response.statusCode', error.code);
+    const statusMessage = get(error, 'response.statusMessage', error.message);
+    let errorMessage = `Failed to ingest, statusCode: ${statusCode}, statusMessage: ${statusMessage}`;
 
-    if (ingestResponseBody.errors) {
-      const xmlObjectError = JSON.stringify(ingestResponseBody.errors.error);
-      throw new Error(`Failed to ingest, CMR error message: ${xmlObjectError}`);
+    if (get(error, 'response.body')) {
+      const parsedResponseBody = <CMRErrorResponseBody>(await parseXMLString(error.response.body));
+      const responseError = get(parsedResponseBody, 'errors.error');
+      if (responseError) {
+        errorMessage = `${errorMessage}, CMR error message: ${JSON.stringify(responseError)}`;
+      }
     }
 
-    return ingestResponseBody;
-  } catch (error) {
-    log.error(error, logDetails);
-    throw error;
+    log.error(errorMessage);
+
+    if (statusCode >= 500 && statusCode < 600) {
+      throw new CMRInternalError(errorMessage);
+    }
+    throw new Error(errorMessage);
```

Comment:
This is nice cleanup, I just can't find where got.put has to throw if it returns an error from the server. (same comment as previous change)

Reply:
I can't find it, but I validated it.